### PR TITLE
Updated plugin GPIO

### PR DESCRIPTION
**Pull request by onkelandy**
Change to GPIO.event_detect to streamline plugin

### CODE REVIEW

File: gpio/__init__.py [hunk 1]
```diff
@@ -79,23 +74,30 @@ def parse_item(self, item):
             GPIO.setup(in_pin, GPIO.IN, pull_up_down=GPIO.PUD_UP)
             self.logger.debug("GPIO: INPUT {0} assigned to pin \'{1}\'".format(item, in_pin))
             self._items.append(item)
+            self._itemsdict[in_pin] = item
             return self.update_item      
         if self.has_iattr(item.conf, 'gpio_out'):
             out_pin = int(self.get_iattr_value(item.conf, 'gpio_out'))
+            GPIO.setup(out_pin, GPIO.IN, pull_up_down=GPIO.PUD_UP)
```
Comment (ohinckel) [inline gpio/__init__.py]: Is this just a copy&paste error or do I need to use `setup()`  with `GPIO.IN` and later with `GPIO.OUT`?

File: gpio/__init__.py [hunk 2]
```diff
@@ -79,23 +74,30 @@ def parse_item(self, item):
...
+            self._itemsdict[out_pin] = item
+            value = GPIO.input(out_pin)
+            item(value, 'GPIO', 'gpio_init')
+            GPIO.add_event_detect(out_pin, GPIO.BOTH, callback=self.get_sensors)
```
Comment (ohinckel) [inline gpio/__init__.py]: I'm not sure, but does this event detection works with output pins too?

File: gpio/__init__.py (hunk 1 above)
Comment (onkelandy) [inline gpio/__init__.py]: The GPIO.IN is necessary to detect value changes in the output pin. Something like a "reply" whether it worked or not

File: gpio/__init__.py (hunk 2 above)
Comment (onkelandy) [inline gpio/__init__.py]: As far as I see it, it works for out pin but I am not totally sure. At least it is possible to query the current value of an output pin with GPIO.input

**Comment by ohinckel** [comment]
Just tested the `gpio_in` with the event decetion implementation: it's working and load is reduced. Thanks for looking into this @onkelandy!

**Pull request merged**